### PR TITLE
Plugin/Action hook PoC, SWC compilation, and basic support for 414 and 498 clients

### PR DESCRIPTION
**Pull request by Kikorono**

### CODE REVIEW

File: src/engine/net/packets/outbound-packets/encoders/window-pane-packet.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ByteBuffer } from '@runejs/common';
+import { OutboundPacket, PacketSize } from '../../packets';
+
+interface WindowPaneData {
+    windowId: number;
```
Comment (boogie-nights) [inline src/engine/net/packets/outbound-packets/encoders/window-pane-packet.ts:5]: we should make a task to fix this

File: src/engine/widgets/widget-config.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { requireConfigFiles } from '@engine/util/fs';
+import { ClientEngineType } from '@engine';
+
+export interface IWidgetConfigData {
+    old_engine_id: number;
```
[the remark this answers is not in the document]
Comment (Kikorono) [reply src/engine/widgets/widget-config.ts:5]: Good thought, I'm not entirely sold on this interface system either, more needed it to test the plugin system

File: src/plugins/logout.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    // ],
+    // compatibleBuilds: [
+    //     289, 319, 357, 414
+    // ]
+})
+export default class LogoutPlugin {
+
+    @ButtonAction([
+        { buttons: 2458 },
+        { widgets: 'rs:logout', buttons: 8 }
```
[the remark this answers is not in the document]
Comment (Kikorono) [reply src/plugins/logout.plugin.ts:20]: Iunno, prolly 😆

File: src/engine/net/packets/outbound-packets/encoders/window-pane-packet.ts (hunk 1 above)
Comment (Kikorono) [reply src/engine/net/packets/outbound-packets/encoders/window-pane-packet.ts:5]: You're a task to fix this

File: src/engine/actions/action.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    handler: HANDLER_FN;
+}
+
+export const Action = <HOOK, HANDLER_FN>(
+    action: IActionHook<HOOK>,
+) => {
+    return function (
+        targetClass: any,
+        propertyKey: string | symbol,
+        // propertyDescriptor: PropertyDescriptor
```
Comment (xlogic1992) [inline src/engine/actions/action.ts:21]: Can we please remove this dead code?

**Review by boogie-nights — APPROVED** [review]

**Review by TylerGC — APPROVED** [review]

**Pull request merged**